sklearn/linear_model/ridge.py

```diff
@@ -845,22 +849,24 @@ def _pre_compute_svd(self, X, y):
         UT_y = np.dot(U.T, y)
         return v, U, UT_y
 
-    def _errors_svd(self, alpha, y, v, U, UT_y):
+    def _errors_and_values_svd_helper(self, alpha, y, v, U, UT_y):
+        """Helper function to avoid code duplication between self._errors_svd
+        and self._values_svd.
+        """
         w = ((v + alpha) ** -1) - (alpha ** -1)
         c = np.dot(U, self._diag_dot(w, UT_y)) + (alpha ** -1) * y
         G_diag = self._decomp_diag(w, U) + (alpha ** -1)
         if len(y.shape) != 1:
             # handle case where y is 2-d
             G_diag = G_diag[:, np.newaxis]
+        return G_diag, c
+
+    def _errors_svd(self, alpha, y, v, U, UT_y):
+        G_diag, c = self._errors_and_values_svd_helper(alpha, y, v, U, UT_y)
         return (c / G_diag) ** 2, c
 
     def _values_svd(self, alpha, y, v, U, UT_y):
-        w = ((v + alpha) ** -1) - (alpha ** -1)
-        c = np.dot(U, self._diag_dot(w, UT_y)) + (alpha ** -1) * y
```

Comment:
@agramfort: look at lines 856--858 of the  `_errors_and_values_svd_helper` function.